### PR TITLE
DSPs prepared for loading persisted models

### PR DESCRIPTION
**Pull request by pawelKapl**
Few changes that will enable loading models from serialized sources:

- added IR snapshot - a small struct that will allow to capture current IR configuration, and then recover IR basing on it
- added new method that allows to create DSP basing on directly passed json config instead of file path. 

This is a first part of the solution for issue sdatkinson/NeuralAmpModelerPlugin#163 second part is here: https://github.com/sdatkinson/NeuralAmpModelerPlugin/pull/206

### CODE REVIEW

**Comment by pawelKapl** [comment]
Hey @sdatkinson did u have a time to take a look at this one? It is now polished, tested well 🤞 , and introduces much less 'breaking changes'. Thanks!

**Comment by sdatkinson** [comment]
Thanks! I'll hopefully have time to look this weekend

File: dsp/ImpulseResponse.h
Comment (sdatkinson) [inline dsp/ImpulseResponse.h:49]: Nit: name `IRData`?

File: dsp/ImpulseResponse.h
Comment (sdatkinson) [inline dsp/ImpulseResponse.h:27]: Nit: name `GetData`?

**Review by sdatkinson — CHANGES_REQUESTED** [review]
Thanks for your patience, getting back to this.

Could you have a look at the changes merged in #46 and see if that resolves your changes in `dsp.h` and `get_dsp.cpp`? I think those changes can be dropped. (This is the part I'm requesting be changed before I can approve

IR part looks good except naming nits; your call whether to grab.

Thanks!

**Comment by pawelKapl** [comment]
@sdatkinson Yeah i think i can marry it without my changes. Dropped them, and changed naming. Now should be fine!

File: dsp/ImpulseResponse.h
Comment (sdatkinson) [inline dsp/ImpulseResponse.h:23]: Can we get this removed? Doesn't seem to be doing anything except hanging out on the public interface?

**Review by sdatkinson — CHANGES_REQUESTED** [review]
Just the one and we're good. Thanks!

**Review by sdatkinson — APPROVED** [review]
LGTM! 👍🏻

**Pull request merged**
